internal/server/iifo.go

```diff
@@ -53,49 +41,41 @@ func (s *SOMASServer) runForageSharing() {
 	s.distributeForageSharing(otherIslandInfo)
 }
 
-func (s *SOMASServer) getPredictions() (shared.PredictionInfoDict, error) {
+func (s *SOMASServer) getPredictions() shared.PredictionInfoDict {
 	islandPredictionsDict := shared.PredictionInfoDict{}
-	var err error
 	nonDeadClients := getNonDeadClientIDs(s.gameState.ClientInfos)
 	for _, id := range nonDeadClients {
 		c := s.clientMap[id]
-		islandPredictionsDict[id], err = c.MakePrediction()
-		if err != nil {
-			return islandPredictionsDict, errors.Errorf("Failed to get prediction from %v: %v", id, err)
-		}
+		islandPredictionsDict[id] = c.MakePrediction()
 	}
-	return islandPredictionsDict, nil
+	return islandPredictionsDict
 }
 
-func (s *SOMASServer) distributePredictions(islandPredictionDict shared.PredictionInfoDict) error {
-	receivedPredictionsDict := make(shared.ReceivedPredictionsDict)
-	var err error
+func (s *SOMASServer) distributePredictions(islandPredictionDict shared.PredictionInfoDict) {
+	reorderDictionary := make(map[shared.ClientID]shared.ReceivedDisasterPredictionsDict)
 	// Add the predictions/sources to the dict containing which predictions each island should receive
 	// Don't allow teams to know who else these predictions were shared with in MVP
+	nonDeadClients := getNonDeadClientIDs(s.gameState.ClientInfos)
+
 	for idSource, info := range islandPredictionDict {
-		for _, idShare := range info.TeamsOfferedTo {
-			if idShare == idSource {
-				continue
+		if clientArrayContains(nonDeadClients, idSource) {
```

Comment:
you only get predictions from non dead clients, does this check need to be done?

Reply:
Not really, but figured it didnt hurt to have it in there